File: pueblo_golf_tournament_mobile/lib/dto/model/tournament-hole-dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pueblo_golf_tournament_mobile/base/base-dto.dart';

part 'tournament-hole-dto.g.dart';

@JsonSerializable()
class TournamentHoleDto extends BaseDto {
  final int tournamentId;
  final int holeNumber;
  final int maximumStrokes;

  TournamentHoleDto(
      {required this.tournamentId,
      required this.holeNumber,
      required this.maximumStrokes,
      required super.id});

  factory TournamentHoleDto.fromJson(Map<String, dynamic> json) =>
      _$TournamentHoleDtoFromJson(json);
  Map<String, dynamic> toJson() => _$TournamentHoleDtoToJson(this);
}

File: pueblo-golf-tournament-api/Dto/TournamentHoleDto.cs
using pueblo_golf_tournament_api.Base;

namespace pueblo_golf_tournament_api.Dto
{
    public class TournamentHoleDto : BaseDto
    {
        public long TournamentId { get; set; }
        public int HoleNumber { get; set; }
        public int MaximumStrokes { get; set; }
    }
}